run/test/specs/user_actions.spec.ts

```diff
@@ -283,64 +302,67 @@ async function setNicknameAndroid(platform: SupportedPlatformsType) {
   // Select username to edit
-  await device1.clickOnElement("Edit user nickname");
+  await device1.clickOnByAccessibilityID("Edit user nickname");
   // Type in nickname
   await device1.inputText("accessibility id", "Username", nickName);
   // Click on tick button
-  await device1.clickOnElement("Apply");
+  await device1.clickOnByAccessibilityID("Apply");
   // CLick out of pop up
-  await device1.clickOnElement("Message user");
+  await device1.clickOnByAccessibilityID("Message user");
   // Check name at top of conversation is nickname
-  const headerElement = await device1.waitForElementToBePresent({
+  const headerElement = await device1.waitForTextElementToBePresent({
     strategy: "accessibility id",
-    selector: "Username",
+    selector: "Conversation header name",
   });
   await device1.getTextFromElement(headerElement);
   // Send a message so nickname is updated in conversation list
   await device1.sendMessage("Howdy");
   // Navigate out of conversation
-  await device1.clickOnElement("Navigate up");
+  await device1.navigateBack(platform);
   // Change nickname back to original username
   // Long press on contact conversation
   await device1.longPressConversation(nickName);
   // Select details
-  await device1.clickOnElement("Details");
+  await device1.clickOnByAccessibilityID("Details");
   // Click on username to edit
-  await device1.clickOnElement("Edit user nickname");
+  await device1.clickOnByAccessibilityID("Edit user nickname");
   // Click apply without entering new nickname
-  await device1.clickOnElement("Apply");
+  await device1.clickOnByAccessibilityID("Apply");
   // Click out of pop up
   await device1.back();
   // Enter conversation to verify change
   await device1.selectByText("Conversation list item", nickName);
-  const changedElement = await device1.waitForElementToBePresent({
+  const changedElement = await device1.waitForTextElementToBePresent({
     strategy: "accessibility id",
-    selector: "Username",
+    selector: "Conversation header name",
   });
-  await device1.getTextFromElement(changedElement);
+  const headerUsername = await device1.getTextFromElement(changedElement);
+  if (headerUsername === nickName) {
+    console.log("Nickname has been changed in header correctly");
+  }
   // Send message to change in conversation list
   await device1.sendMessage("Howdy");
   // Navigate back to list
-  await device1.clickOnElement("Navigate up");
+  await device1.navigateBack(platform);
   // Verify name change in list
   // Save text of conversation list item?
-  await device1.selectByText("Conversation list item", userB.userName);
-  const revertedHeader = await device1.waitForElementToBePresent({
+  await device1.selectByText("Conversation list item", nickName);
+  const changedListName = await device1.waitForTextElementToBePresent({
     strategy: "accessibility id",
-    selector: "Username",
+    selector: "Conversation header name",
+    text: nickName,
```

Comment:
naming isn't great here, changedListName, listName, and nickname.
it's ok  for the last 2, but changedListName is actually an element, so maybe listNameElement instead?